dash-pipeline/SAI/templates/utils.cpp.j2

```diff
@@ -94,38 +96,46 @@ grpc::StatusCode MutateTableEntry(std::shared_ptr<p4::v1::TableEntry> entry, p4:
     return status.error_code();
 }
 
-bool InsertInTable(std::shared_ptr<p4::v1::TableEntry> entry, sai_object_id_t *objId)
+bool InsertInTable(std::shared_ptr<p4::v1::TableEntry> entry, sai_object_type_t objectType, sai_object_id_t *objId)
 {
     DASH_LOG_ENTER();
 
     auto retCode = MutateTableEntry(entry, p4::v1::Update_Type_INSERT);
-    if (grpc::StatusCode::OK != retCode) {
+
+    if (grpc::StatusCode::OK != retCode)
+    {
         return false;
     }
 
-    std::lock_guard<std::mutex> lock(tableLock);
+    MUTEX;
 
     if (*objId == 0)
     {
-        *objId = NextObjIndex();
+        *objId = NextObjectId(objectType);
     }
+
     tableEntryMap.insert(std::make_pair(*objId, entry));
+
     return true;
 }
 
-sai_object_id_t NextObjIndex()
+sai_object_id_t NextObjectId(sai_object_type_t objectType)
 {
     DASH_LOG_ENTER();
 
+    // TODO make proper oid
```

Comment:
Don't you want to address this now as well?

Reply:
addressed